### PR TITLE
Handle "pair of x" when x in `pl_sb_uninflected_complete`

### PR DESCRIPTION
**Pull request by paul2048**
Issue raised here: https://github.com/jaraco/inflect/issues/184

### CODE REVIEW

File: inflect/__init__.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def make_pl_si_lists(
     "ox": "oxen",
     "cow": "cows|kine",
     "graffito": "graffiti",
-    "octopus": "octopuses|octopodes",
+    "octopus": "octopuses|octopodes|octopi",
```
Comment (jaraco) [inline inflect/__init__.py:222]: This change is unrelated and [technically wrong](https://oceanconservancy.org/blog/2022/02/01/plural-octopus/). I'd prefer to omit it, but if you want to advocate for adding it, let's do that in a separate issue/PR.

File: inflect/__init__.py [hunk 2]
```diff
@@ -2715,6 +2715,14 @@ def _plnoun(  # noqa: C901
...
 
         if word.last.lower() in pl_sb_uninflected_complete:
+            if len(word.split_) >= 3:
+                for numword in range(1, len(word.split_) - 1):
+                    if word.split_[numword] in pl_prep_list_da:
+                        return " ".join(
+                            word.split_[: numword - 1]
+                            + [self._plnoun(word.split_[numword - 1], 2)]
+                            + word.split_[numword:]
+                        )
```
Comment (jaraco) [inline inflect/__init__.py:2725]: This change adds a lot of complexity to this function, with multiple nested conditionals and loops. This method (`_plnoun`) is already tagged with `noqa: C901` because it's already too complex. Let's not make it more complex. I'd like to explore a way to encapsulate this logic in a method such that it can be documented and compartmentalized.

File: inflect/__init__.py [hunk 3]
```diff
@@ -3201,6 +3209,14 @@ def _sinoun(  # noqa: C901
...
 
         if words.last.lower() in pl_sb_uninflected_complete:
+            if len(words.split_) >= 3:
+                for numword in range(1, len(words.split_) - 1):
+                    if words.split_[numword] in pl_prep_list_da:
+                        return " ".join(
+                            words.split_[: numword - 1]
+                            + [self._sinoun(words.split_[numword - 1], 1)]
+                            + words.split_[numword:]
+                        )
```
Comment (jaraco) [inline inflect/__init__.py:3219]: This looks like a copy/paste of other code. Although existing `inflect` logic may have used copy/paste as a way to re-use behavior, I'd like to avoid it for future contributions. Let's either first refactor the common behavior of `_sinoun` and `_plnoun` so that the change is only needed once or at the very least provide this functionality as its own method so that only one line needs to be repeated in both methods. I acknowledge that may prove difficult given the nested conditionals, loops, and inner return. Let me know if you'd like my help with that.

**Review by jaraco — CHANGES_REQUESTED** [review]
Thanks for the contrib. I appreciate that you've provided a test and an implementation. I have a few suggestions I'd like addressed. Please address these suggestions or let me know if you'd like help.

File: inflect/__init__.py (hunk 1 above)
Comment (paul2048) [reply inflect/__init__.py:222]: I agree! Let's omit it in this case

File: inflect/__init__.py (hunk 2 above)
Comment (paul2048) [reply inflect/__init__.py]: Addressed: https://github.com/jaraco/inflect/pull/188/commits/712906cc85bf0942fd540263965c104936b44f9d
I just can't figure out a way to add this
```
        >>> engine._handle_long_compounds(self, "pair of scissors", 2)
        'pairs of scissors'
        >>> engine._handle_long_compounds(self, "pairs of scissors", 1)
        'pair of scissors'
        >>> engine._handle_long_compounds(self, "pound per square inch", 2)
        'pounds per square inch'
```
in the the `_handle_long_compounds`'s docstring. It expects the `self` argument which we don't pass when we call it in `_plnoun` and `_sinoun`...

File: inflect/__init__.py (hunk 3 above)
Comment (paul2048) [reply inflect/__init__.py]: Thanks for the help offering! I came up with this, so let me know what you think: https://github.com/jaraco/inflect/pull/188/commits/712906cc85bf0942fd540263965c104936b44f9d

Comment (jaraco) [reply inflect/__init__.py]: Much better. Thanks!

File: inflect/__init__.py (hunk 2 above)
Comment (jaraco) [reply inflect/__init__.py]: Try `engine()._handle_long_compounds("pair of scissors", 2)`. Since it's an instance method. Or, you could make it a `@classmethod` and call it without creating an instance (`engine()._handle_long_compounds("pair of scissors", 2)`.

Comment (paul2048) [reply inflect/__init__.py]: Thanks! Addressed here: https://github.com/jaraco/inflect/pull/188/commits/76619bf16a59cb5d3c8e6b687f54ce7fb688fa91

**Comment by paul2048** [comment]
Hey @jaraco, does everything look good now?
